### PR TITLE
VS2022 Win10 build fix

### PR DESCRIPTION
**Pull request by vovodroid**
- use DEPS variable for both deps and slicer build
- use installed WinSDK version
- echo real cmake command line

### CODE REVIEW

File: build_release_vs2022.bat [hunk 1]
```diff
@@ -44,9 +44,10 @@ if "%1"=="slicer" (
 )
 echo "building deps.."
 
-echo cmake ../ -G "Visual Studio 17 2022" -A x64 -DDESTDIR="%CD%/OrcaSlicer_dep" -DCMAKE_BUILD_TYPE=%build_type% -DDEP_DEBUG=%debug% -DORCA_INCLUDE_DEBUG_INFO=%debuginfo%
-cmake ../ -G "Visual Studio 17 2022" -A x64 -DDESTDIR="%CD%/OrcaSlicer_dep" -DCMAKE_BUILD_TYPE=%build_type% -DDEP_DEBUG=%debug% -DORCA_INCLUDE_DEBUG_INFO=%debuginfo%
+echo on
```
Comment (Ocraftyone) [inline build_release_vs2022.bat:47]:
```suggestion
@echo on
```

File: build_release_vs2022.bat [hunk 2]
```diff
@@ -56,9 +57,10 @@ cd %WP%
 mkdir %build_dir%
 cd %build_dir%
 
-echo cmake .. -G "Visual Studio 17 2022" -A x64 -DBBL_RELEASE_TO_PUBLIC=1 -DCMAKE_PREFIX_PATH="%DEPS%/usr/local" -DCMAKE_INSTALL_PREFIX="./OrcaSlicer" -DCMAKE_BUILD_TYPE=%build_type%
-cmake .. -G "Visual Studio 17 2022" -A x64 -DBBL_RELEASE_TO_PUBLIC=1 -DCMAKE_PREFIX_PATH="%DEPS%/usr/local" -DCMAKE_INSTALL_PREFIX="./OrcaSlicer" -DCMAKE_BUILD_TYPE=%build_type% -DWIN10SDK_PATH="C:/Program Files (x86)/Windows Kits/10/Include/10.0.22000.0"
+echo on
```
Comment (Ocraftyone) [inline build_release_vs2022.bat:60]:
```suggestion
@echo on
```

File: build_release_vs2022.bat [hunk 3]
```diff
@@ -56,9 +57,10 @@ cd %WP%
...
+cmake .. -G "Visual Studio 17 2022" -A x64 -DBBL_RELEASE_TO_PUBLIC=1 -DCMAKE_PREFIX_PATH="%DEPS%/usr/local" -DCMAKE_INSTALL_PREFIX="./OrcaSlicer" -DCMAKE_BUILD_TYPE=%build_type% -DWIN10SDK_PATH="%WindowsSdkDir%Include\%WindowsSDKVersion%\"
```
Comment (Ocraftyone) [inline build_release_vs2022.bat:61]: Where is the Windows SDK path/version being gotten from? If you are pulling it from the ENV variables set by CMake for the deps build, remember that the slicer can be build independently of the deps and vice versa

File: build_release_vs2022.bat (hunk 1 above)
Comment (vovodroid) [reply build_release_vs2022.bat:47]: echo anyway is off at this line

File: build_release_vs2022.bat (hunk 2 above)
Comment (vovodroid) [reply build_release_vs2022.bat:60]: echo anyway is off at this line

File: build_release_vs2022.bat (hunk 3 above)
Comment (vovodroid) [reply build_release_vs2022.bat:61]: It's from x64 Native tool window environment, so it's set before BAT file is run.

File: build_release_vs2022.bat (hunk 2 above)
Comment (Ocraftyone) [reply build_release_vs2022.bat:60]: I see. Nvmd

File: build_release_vs2022.bat (hunk 1 above)
Comment (Ocraftyone) [reply build_release_vs2022.bat:47]: I see. Nvmd

File: build_release_vs2022.bat (hunk 3 above)
Comment (Ocraftyone) [reply build_release_vs2022.bat:61]: Only problem is if you don't run from that environment. I typically run that script from the CLion terminal which results in the path not being set correctly. Maybe check if the variables are initialized and if not set the current values as default?

**Comment by Ocraftyone** [comment]
I understand the convenience having the command echo itself rather than having the command written out in the script twice, but I personally like the look of not having the user path before the command. I do have to admit that is just me being very nitpicky though 😄

File: build_release_vs2022.bat (hunk 3 above)
Comment (vovodroid) [reply build_release_vs2022.bat:61]: Actually build instruction clearly says `Run build_release.bat in x64 Native Tools Command Prompt for VS 2019` (applied to 2022 as well I guess).

**Comment by vovodroid** [comment]
> I personally like the look of not having the user path before the command.

It was included into echo before this PR, so I just eliminated copy-paste errors.

**Review by SoftFever — APPROVED** [review]
Nice change
Thank you

**Comment by Noisyfox** [comment]
This breaks build on Github Actions: #6724
If you look at the build log https://github.com/SoftFever/OrcaSlicer/actions/runs/10795985355/job/29943964894
image
`%WindowsSdkDir%` and `%WindowsSDKVersion%` apparently is missing in that environment.

**Pull request merged**